src/api/files.js

```diff
@@ -0,0 +1,25 @@
+'use strict'
+
+const argCommand = require('../cmd-helpers').argCommand
+
+module.exports = send => {
+  return {
+    cp: argCommand(send, 'files/cp'),
+    ls: argCommand(send, 'files/ls'),
+    mkdir: argCommand(send, 'files/mkdir'),
+    stat: argCommand(send, 'files/stat'),
+    rm: function (path, opts, cb) {
```

Comment:
@whyrusleeping can give more input here but if you have

```bash
/folder -> hash Qmabcdef
  fileA

# and then add one more file, the folder MerkleDAG node will update

/folder -> hash Qmewasdajl
  fileA
  fileB

# if you decide to rm Qmabcdef, you might not want to delete it recursively because it might delete fileA that it was pointing to
```